internal/pkg/service/common/servicectx/servicectx_test.go

```diff
@@ -15,71 +16,83 @@ import (
 func TestProcess_Add(t *testing.T) {
 	t.Parallel()
 
-	ctx, cancel := context.WithCancel(context.Background())
 	logger := log.NewDebugLogger()
-	proc, err := New(ctx, cancel, WithLogger(logger), WithUniqueID("<id>"))
+	logger.ConnectTo(os.Stdout)
+	proc, err := New(WithLogger(logger), WithUniqueID("<id>"))
 	assert.NoError(t, err)
 
-	op1 := &sync.WaitGroup{}
-	op1.Add(1)
-	op2 := &sync.WaitGroup{}
-	op2.Add(1)
-	op3 := &sync.WaitGroup{}
-	op3.Add(1)
-
-	// Do some work, operations run in parallel
-	proc.Add(func(ctx context.Context, shutdown ShutdownFn) {
-		<-ctx.Done()
-		logger.InfoCtx(ctx, "end1")
-		op1.Done()
+	// OpCtx simulates long running operations
+	opCtx, opCancel := context.WithCancel(context.Background())
+
+	// There are some parallel operations
+	opWg := &sync.WaitGroup{}
+	opWg.Add(3)
+	proc.Add(func(shutdown ShutdownFn) {
```

Comment:
Test has been modified:
- `Process.Add` goroutines run in parallel, this test tests it (second test `TestProcess_Shutdown` is unmodified).